lib/parse.js

```diff
@@ -334,9 +334,8 @@ function validateRange(state) {
 }
 
 function isMultipleOf(value, step) {
-  const decimal = step.toString().split('.')[1];
-  const scale = decimal ? Math.pow(10, decimal.length) : 1;
-  return value * scale % (step * scale) === 0;
+  const n = value / step;
+  return n - Math.floor(n) < 1e-15; // IEEE 754 numbers can reliably store only 15 digits
```

Comment:
It should be `Math.abs(n - Math.round(n))` e.g. when `n = 0.3/0.1`.

For the epsilon, TBH I don't have enough knowledge to choose a correct one.

Reply:
Fixed.

As for the epsilon, there's one case where it's unclear what isMultipleOf should do: when the value has more than 15 meaningful digits. Currently the function returns true for any `step` because the result of division lacks the necessary precision to make a decision.